scripts/build_test.py

```diff
@@ -14,29 +14,313 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import StringIO
+import collections
+import os
+import random
+import re
+import subprocess
+import threading
+
 # pylint: disable=relative-import
 import build
 from core.tests import test_utils
 
 # pylint: enable=relative-import
 
+TEST_DIRECTORY = os.path.join('core', 'tests', 'build')
+TEST_SOURCE_DIRECTORY = os.path.join('core', 'tests', 'build_sources')
+
+ASSETS_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'assets', '')
+ASSETS_OUT_DIR = os.path.join(TEST_DIRECTORY, 'static', 'assets', '')
+
+EXTENSIONS_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'extensions', '')
+
+TEMPLATES_DEV_DIR = os.path.join(TEST_SOURCE_DIRECTORY, 'templates', '')
+
```

Comment:
So the `''` at the end of these os.path variables makes them directory path: i.e.
```Python
os.path.join('static', 'assets', '')
``` 
is `/static/assets/`
vs
```Python
os.path.join('static', 'assets')
``` 
will be `/static/assets`, which os.walk would not accept as a valid dir path.